logging/logrus/middleware.go

```diff
@@ -35,8 +35,12 @@ func Middleware(entry *logrus.Entry, opts ...Option) httpwares.Middleware {
 			})
 			startTime := time.Now()
 			nextHandler.ServeHTTP(wrappedResp, newReq)
-			capture.finish() // captureResponse has a nil check, this can be nil
 
+			if !o.shouldLog(wrappedResp, newReq) {
```

Comment:
if you do `if o.shouldLog != nil && !o.shouldLog(....) {...}` then you won't need to have the default decider.